### PR TITLE
Give nicer error to users when loading modules from different toolchains

### LINKED ISSUE #9: better error message when loading conflicting modules
Users now get to see something like this, which is confusing them:

```
$ ml Python/2.7.11-intel-2016a

$ ml R/3.3.1-intel-2016b
Lmod has detected the following error:  Your site prevents the automatic swapping of modules with same name. You must explicitly unload the loaded version of "intel" before you can load the new
one. Use swap (or an unload followed by a load) to do this:

   $ module swap intel  intel/2016b

Alternatively, you can set the environment variable LMOD_DISABLE_SAME_NAME_AUTOSWAP to "no" to re-enable same name autoswapping. 
While processing the following module(s):
    Module fullname      Module Filename
    ---------------      ---------------
    R/3.3.1-intel-2016b  /apps/gent/SL6/sandybridge/modules/all/R/3.3.1-intel-2016b

If you don't understand the warning or error, contact the helpdesk at hpc@ugent.be
```

First of all, the swapping suggestion is confusing since it won't get them out of the woods.

@wpoely86 Can we provide a more meaningful error message here via a hook?

### PR DESCRIPTION
**Pull request by wpoely86**
Fixes #9 

With this:

```
$ ml zlib/1.2.8-intel-2014b
$ ml ncurses
Lmod has detected the following error:  You can not load modules using different toolchains. A different version of the 'intel' module is already loaded (see output of 'ml').
You should load another 'ncurses' module for that is compatible with the currently loaded version of 'intel'. Use 'ml spider ncurses' to get an overview of the available versions. 

While processing the following module(s):
    Module fullname          Module Filename
    ---------------          ---------------
    ncurses/6.0-intel-2016a  /apps/gent/SL6/westmere/modules/all/ncurses/6.0-intel-2016a

If you don't understand the warning or error, contact the helpdesk at hpc@ugent.be

```

### CODE REVIEW

File: SitePackage.lua [hunk 1]
```diff
@@ -132,6 +132,26 @@ local function msg_hook(mode, output)
...
+    if output[1] and output[1]:find("Your site prevents the automatic swapping of modules with same name") then
+        -- find the module name causing the issue (almost always toolchain module)
+        local mname
+        mname = output[1]:match("$ module swap [^ ]+%s+([^ \n]+)")
+        local mStack   = ModuleStack:moduleStack()
+
+        local label  = colorize("red", "Lmod has detected the following error: ")
+        local twidth = TermWidth()
+        local s      = {}
+        s[#s+1] = buildMsg(twidth, label, "You can not load modules using different toolchains.",
```
Comment (boegel) [inline SitePackage.lua]: @wpoely86 this error will not only pop up when two versions of the same toolchain but different version is loaded, but also for other conflicts on module name, right?

For example, if someone loads A which depends on Python 2.7.12, and then loads B which depends on Python 3.5.2?

So, we should not mention 'toolchian' at all in the error message, and only talk about 'module name' and 'module version'?

**Comment by wpoely86** [comment]
What does this do:

```
$ ml zlib/1.2.8-intel-2015a
$ ml zlib/1.2.8-intel-2016a
Lmod has detected the following error:  A different version of the 'zlib' module is already loaded (see output of 'ml'). 



If you don't understand the warning or error, contact the helpdesk at hpc@ugent.be

$ ml ncurses/6.0-intel-2016a
Lmod has detected the following error:  A different version of the 'intel' module is already loaded (see output of 'ml'). You should load another 'ncurses' module for that is compatible with the currently loaded version of
'intel'. Use 'ml spider ncurses' to get an overview of the available versions. 

While processing the following module(s):
    Module fullname          Module Filename
    ---------------          ---------------
    ncurses/6.0-intel-2016a  /apps/gent/SL6/westmere/modules/all/ncurses/6.0-intel-2016a

If you don't understand the warning or error, contact the helpdesk at hpc@ugent.be

```

File: SitePackage.lua [hunk 2]
```diff
@@ -132,6 +132,26 @@ local function msg_hook(mode, output)
...
+    if output[1] and output[1]:find("Your site prevents the automatic swapping of modules with same name") then
+        -- find the module name causing the issue (almost always toolchain module)
+        local sname  = output[1]:match("$ module swap ([^ ]+)%s+[^ \n]+")
+        local mStack = ModuleStack:moduleStack()
+
+        local errmsg = {"A different version of the '"..sname.."' module is already loaded (see output of 'ml')."}
+        if not mStack:empty() then
+            errmsg[#errmsg+1] = " You should load another '"..mStack:sn().."' module for that is compatible with the "
+            errmsg[#errmsg+1] = "currently loaded version of '"..sname.."'. Use 'ml spider "..mStack:sn().."' to get "
+            errmsg[#errmsg+1] = "an overview of the available versions."
```
Comment (boegel) [inline SitePackage.lua]: @wpoely86 start every sentence on a new line, to try and make sure people will actually _read_

Comment (wpoely86) [reply SitePackage.lua]: The terminal does the wrapping, not me.

Comment (boegel) [reply SitePackage.lua]: I know the terminal wraps, but starting each new sentence on a new line will significantly improve the readability of the error message. Try it, and compare.

**Comment by wpoely86** [comment]
Now:

```
$ ml GSL
Lmod has detected the following error:  A different version of the 'intel' module is already loaded (see output of 'ml').
You should load another 'GSL' module for that is compatible with the currently loaded version of 'intel'.
Use 'ml spider GSL' to get an overview of the available versions. 

While processing the following module(s):
    Module fullname      Module Filename
    ---------------      ---------------
    GSL/2.1-intel-2016b  /apps/gent/SL6/sandybridge/modules/all/GSL/2.1-intel-2016b

If you don't understand the warning or error, contact the helpdesk at hpc@ugent.be

```

**Comment by boegel** [comment]
@wpoely86 looks good, but can we avoid the excessive vertical whitespace when loading a module that is already loaded?

```
[11:30:46] vsc40023@test2802:~/vsc-testing/module $ ml Python/2.7.9-intel-2015a 
[11:30:56] vsc40023@test2802:~/vsc-testing/module $ ml 

Currently Loaded Modules:
  1) cluster/.banette (S)   3) icc/2015.1.133-GCC-4.9.2     5) iccifort/2015.1.133-GCC-4.9.2                  7) iimpi/7.2.3-GCC-4.9.2                   9) intel/2015a              11) zlib/1.2.8-intel-2015a       13) ncurses/5.9-intel-2015a
  2) GCC/4.9.2              4) ifort/2015.1.133-GCC-4.9.2   6) impi/5.0.2.044-iccifort-2015.1.133-GCC-4.9.2   8) imkl/11.2.1.133-iimpi-7.2.3-GCC-4.9.2  10) bzip2/1.0.6-intel-2015a  12) libreadline/6.3-intel-2015a  14) Python/2.7.9-intel-2015a

  Where:
   S:  Module is Sticky, requires --force to unload or purge
[11:30:58] vsc40023@test2802:~/vsc-testing/module $ ml Python/3.4.3-intel-2015a
Lmod has detected the following error:  A different version of the 'Python' module is already loaded (see output of 'ml'). 



If you don't understand the warning or error, contact the helpdesk at hpc@ugent.be

[11:31:04] vsc40023@test2802:~/vsc-testing/module $ ml matplotlib/1.4.3-intel-2015a-Python-3.4.3 
Lmod has detected the following error:  A different version of the 'Python' module is already loaded (see output of 'ml').
You should load another 'matplotlib' module for that is compatible with the currently loaded version of 'Python'.
Use 'ml spider matplotlib' to get an overview of the available versions. 

While processing the following module(s):
    Module fullname                            Module Filename
    ---------------                            ---------------
    matplotlib/1.4.3-intel-2015a-Python-3.4.3  /apps/gent/SL6/westmere/modules/all/matplotlib/1.4.3-intel-2015a-Python-3.4.3

If you don't understand the warning or error, contact the helpdesk at hpc@ugent.be

[11:32:01] vsc40023@test2802:~/vsc-testing/module $ 
```

File: Lmod-UGent.spec [hunk 3]
```diff
@@ -1,7 +1,7 @@
 %global macrosdir %(d=%{_rpmconfigdir}/macros.d; [ -d $d ] || d=%{_sysconfdir}/rpm; echo $d)
 
 Name:           Lmod
-Version:        6.5.10
+Version:        6.5.11
```
Comment (boegel) [inline Lmod-UGent.spec]: @wpoely86 please bump to 6.5.12 (cfr. https://github.com/TACC/Lmod/issues/162)

**Review by boegel — CHANGES_REQUESTED** [review]

**Comment by boegel** [comment]
@wpoely86 ping on avoiding the excessive vertical whitespace?

**Comment by wpoely86** [comment]
Now:

```
$ ml zlib/1.2.8-intel-2015a
$ ml zlib/1.2.8-intel-2015b
Lmod has detected the following error:  A different version of the 'zlib' module is already loaded (see output of 'ml'). 

If you don't understand the warning or error, contact the helpdesk at hpc@ugent.be

$ ml ncurses/5.9-intel-2014b
Lmod has detected the following error:  A different version of the 'intel' module is already loaded (see output of 'ml').
You should load another 'ncurses' module for that is compatible with the currently loaded version of 'intel'.
Use 'ml spider ncurses' to get an overview of the available versions. 
While processing the following module(s):
    Module fullname          Module Filename
    ---------------          ---------------
    ncurses/5.9-intel-2014b  /apps/gent/SL6/westmere/modules/all/ncurses/5.9-intel-2014b

If you don't understand the warning or error, contact the helpdesk at hpc@ugent.be

```

**Review by boegel — APPROVED** [review]

**Comment by boegel** [comment]
lgtm

**Pull request merged**